protocols/raft/src/main/java/io/atomix/protocols/raft/partition/RaftPartition.java

```diff
@@ -300,6 +293,10 @@ public RaftPartitionServer getServer() {
   }
 
   private void onFailure() {
-    server.stop();
+    CompletableFuture.allOf(
+            raftFailureListeners.stream()
+                .map(RaftFailureListener::onRaftFailed)
+                .toArray(CompletableFuture[]::new))
+        .join();
```

Comment:
This can throw an exception - can we end up in a loop again of handling uncaught exceptions?

Reply:
No. `RaftContext::notifyFailureListenener` catches Exception and logs it.